modules/parsers/nasm/nasm-parse.c

```diff
@@ -579,6 +580,34 @@ parse_exp(yasm_parser_nasm *parser_nasm)
 
             yasm_dvs_initialize(&dvs);
             for (;;) {
+                if (curtok == STRING_OP) {
+                    yasm_utfenc enc = STRING_val.enc;
+                    get_next_token();
+                    if (curtok != '(') {
+                        yasm_error_set(YASM_ERROR_SYNTAX,
+                                       N_("expected ( after string op identifier"));
+                        return NULL;
+                    }
+                    get_next_token();
+                    if (curtok != STRING) {
+                        yasm_error_set(YASM_ERROR_SYNTAX,
+                                       N_("expected string argument to string op, got %s"),
+                                       describe_token(curtok));
+                        return NULL;
+                    }
+                    dv = yasm_dv_create_string(STRING_val.contents, STRING_val.len, enc);
+                    if (!dv) {
+                        yasm_error_set(YASM_ERROR_GENERAL, N_("error encoding utf"));
+                    }
+                    get_next_token();
+                    if (curtok != ')') {
```

Comment:
is there a better or more concise way to express parsing rules of `'(' STRING ')'` than this series `get_next_token` and check for errors?